File: lib-cli/src/workflow/render/producer/typescript/static/events/disconnected.ts
import {
	Identification,
	Filter,
	Producer,
	Context,
	Protocol,
} from "../implementation/events";
import { Output } from "../implementation/events/disconnected";

export function emit(
	consumer: Identification,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<Output> {
	return Promise.reject(
		new Error(`Handler for event "disconnected" isn't implemented`)
	);
}

File: experimental/workflow_mt/producer/rust/src/producer/implementation/emitters/disconnected.rs
use super::{
    broadcast, events, hub, identification, pack, producer::Control, Context, EmitterError,
};
use clibri::server;
use uuid::Uuid;

pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    identification: &identification::Identification,
    filter: hub::filter::Filter,
    context: &Context,
    control: &Control<E, C>,
) -> Result<(), EmitterError> {
    events::disconnected::emit::<E, C>(identification, filter, context, control)
        .await
        .map_err(EmitterError::Emitting)
}
